wallet/wallet_v4.tact

```diff
@@ -0,0 +1,121 @@
+interface Plugin {
+    internal requested_funds#f06c7567(query_id: Uint64);
+    internal notify_removed#e4737472(query_id: Uint64);
+    internal notify_installed#6e6f7465(query_id: Uint64);
+    internal notify_removed#64737472(query_id: Uint64);
+}
+
+contract Wallet {
+    var seqno: Uint32
+    var subwallet: Uint32
+    var pubkey: Pubkey
+    var plugins: HashSet[Plugin]
+
+    bounced default() {
+        // ignore bounced messages
+    }
+
+    internal default() {
+        // ignore simple transfers
+    }
+
+    internal request_funds#706c7567(
+        query_id: Uint64,
+        requested_amount: Coins,
+        requested_extra: ExtraCurrencies,
+        msg.sender: Plugin,
+    ) {
+        verify(self.balance() - msg.amount >= requested_amount, "Not enough funds");
```

Comment:
First of all: we must verify that the sender is actually our plugin and not some random contract.

Then we could probably write the balance condition clearly:

```suggestion
        // We subtract the incoming amount because it's already included in the balance
        // and we are sending it back alongside the requested_amount.
        verify(requested_amount <= self.balance()-msg.amount, "Not enough funds");
```

Reply:
Fixed in [090d6cf](https://github.com/tact-lang/tact-bikeshed/pull/6/commits/090d6cfd46d44fe9e117a26fae67de0fcbd36ec5) and [f3baa85](https://github.com/tact-lang/tact-bikeshed/pull/6/commits/f3baa8580c0fb24e61af750ee7bf7f7a9b5585bc)